files/file_helpers.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) F5, Inc.
+//
+// This source code is licensed under the Apache License, Version 2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+
+// Package files implements utility routines for gathering information about files nad their contents.
+package files
+
+import (
+	"cmp"
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"os"
+	"slices"
+
+	"github.com/nginx/agent/v3/api/grpc/mpi/v1"
+	"github.com/nginx/agent/v3/internal/uuid"
+)
+
+// GetPermissions returns a file's permissions as a string.
+func GetPermissions(fileMode os.FileMode) string {
+	return fmt.Sprintf("%#o", fileMode.Perm())
+}
+
+// GenerateConfigVersion returns a unique config version for a set of files.
+// The config version is calculated by joining the file hashes together and generating a unique ID.
+func GenerateConfigVersion(fileSlice []*v1.File) string {
+	var hashes string
+
+	slices.SortFunc(fileSlice, func(a, b *v1.File) int {
+		return cmp.Compare[string](a.GetFileMeta().GetName(), b.GetFileMeta().GetName())
```

Comment:
nit: my IDE was saying the type argument is unnecessary

<img width="1560" alt="image" src="https://github.com/nginx/agent/assets/85518764/be5d1a1f-b539-49b2-93fc-bd7c167ef292">

```suggestion
		return cmp.Compare(a.GetFileMeta().GetName(), b.GetFileMeta().GetName())
```